tools/reactive-controllers/src/PendingState.ts

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2024 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { html, ReactiveController, ReactiveElement, TemplateResult } from 'lit';
+import('@spectrum-web-components/progress-circle/sp-progress-circle.js');
+
+/**
+ * Represents a host element with pending state.
+ */
+export interface HostWithPendingState extends ReactiveElement {
+    pendingLabel?: string;
```

Comment:
Why are you extending ReactiveElement here?

Reply:
I believe both SpectrumElement and ReactiveElement should work